dev/tests/static/testsuite/Magento/Test/Legacy/ObsoleteCodeTest.php

```diff
@@ -210,7 +210,7 @@ protected function _testObsoleteClasses($content)
         foreach (self::$_classes as $row) {
             list($class, , $replacement) = $row;
             $this->_assertNotRegExp(
-                '/[^a-z\d_]' . preg_quote($class, '/') . '[^a-z\d_\\\\]/iS',
+                '/[^a-z\d_\$]' . preg_quote($class, '/') . '[^a-z\d_\\\\]/iS',
```

Comment:
could you explain this change?

Reply:
Hello @ihor-sviziev 
Thank you for your question.
This part of the `legacy` test was incorrect.
We had a problem in class `Magento\Framework\Webapi\Rest\Request\Deserializer\Xml` with param in constructor `$xmlParser` because `XMLParser` class is obsolete.
So, the regular expression was not fully correct.
We fixed it.